### ISSUE TITLE
Accessibility: `ToggleCompositeField` doesn't follow accessibility best practices

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/.github#470]
There are various accessibility problems with the `ToggleCompositeField` right now:

- Uses an interactable `h4` element instead of a `button` element
- Includes a `<a href="#">` element inside the `h4` - but since the `h4` is tabbable the link is redundant
- Missing a bunch of appropriate `aria` attributes to indicate to screen reader users what they're focused on and what it does when activated (`aria-expanded` etc)

Probably makes sense to follow the [W3C accordion pattern](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/) for this. There's also the [W3C Disclosure pattern](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/) which is similar.

## Related

- https://github.com/silverstripe/silverstripe-admin/issues/1965
